### PR TITLE
fix(UIShell): replace "javascript:void(0)" href value in HeaderMenu

### LINKED ISSUE #4353: [React]: "javascript:void(0)" href value in HeaderMenu throws React warning
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

For the [`HeaderMenu` component](https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/UIShell/HeaderMenu.js#L183) in UIShell, a React warning is emitted to the console as an error. The warning reads that a future React version will block javascript URLs (i.e. "javascript:void(0)").

Replacing "javascript:void(0)" with "#" would remove the warning.

## Steps to reproduce the issue

1. Visit Reduced Test case in [CodeSandbox](https://codesandbox.io/s/codesandbox-xfnip)
2. Open the console
3. React warning is printed in the console as an error

## Additional information

React warning

### PR DESCRIPTION
**Pull request by metonym**
Closes #4353 

This PR replaces the "javascript:void(0)" href value with "#" in HeaderMenu to remove the React warning message.

#### Changelog

**New**

- prevent default in `handleOnClick` method

**Changed**

- replace "javascript:void(0)" with "#"

#### Testing / Reviewing

In the React workspace, run `yarn start` to start the storybook locally. Navigate to the [UI Shell Header Base w/ Navigation story](http://localhost:9000/?path=/story/ui-shell--header-base-w-navigation). Open the console; the React warning should not be emitted.

### CODE REVIEW

File: packages/react/src/components/UIShell/HeaderMenu.js [hunk 1]
```diff
@@ -72,7 +72,9 @@ class HeaderMenu extends React.Component {
   /**
    * Toggle the expanded state of the menu on click.
    */
-  handleOnClick = () => {
+  handleOnClick = e => {
+    e.preventDefault();
```
Comment (joshblack) [inline packages/react/src/components/UIShell/HeaderMenu.js:76]: Should we preventing the default here?

**Review by joshblack — COMMENTED** [review]
Definitely appreciate the update! Left one question above

File: packages/react/src/components/UIShell/HeaderMenu.js (hunk 1 above)
Comment (metonym) [reply packages/react/src/components/UIShell/HeaderMenu.js:76]: I could go either way, but I feel that preventing the default would be expected, especially to preserve a cleaner url (i.e. no "#" appended).

**Comment by joshblack** [comment]
cc @dakahn are there any UI Shell issues that we have around this pattern for the HeaderMenu in terms of accessibility? Under the hood it was using the menubar pattern but I think we're nixing that right?

**Review by dakahn — APPROVED** [review]

**Comment by dakahn** [comment]
@joshblack Yeah, we're going to be rethinking and tinkering with accessibility and the UIShell in a future work cycle. We're definitely not looking for a OS style menu keyboard interaction in the future though.

**Review by joshblack — APPROVED** [review]

**Review by abbeyhrt — APPROVED** [review]

**Pull request merged**